manual/manual/tutorials/coreexamples.etex

```diff
@@ -677,6 +677,90 @@ print_expr e; print_newline ();;
 print_expr (deriv e "x"); print_newline ();;
 \end{caml_example}
 
+\section{Printf formats}
+
+There is a "printf" function in
+\href{https://caml.inria.fr/pub/docs/manual-ocaml/libref/Printf.html}{Printf}
+module (see chapter~\ref{c:moduleexamples}) that allows you to make formatted
+output more concisely.
+It follows the behavior of the "printf" function from the C standard library.
+The "printf" function takes a format string that describes the output values and
+their format, and arguments for output that correspond to the formatting
+directives in the format string.
+\begin{caml_example}{toplevel}
+Printf.printf "%i + %i is an integer value, %F * %F is a float, %S\n"
+3 2 4.5 1. "this is a string";;
+\end{caml_example}
+The OCaml type system checks that the type of the arguments and the specifiers are
+compatible. If you pass it an argument of a type that does not correspond to
+the format specifier, the compiler will display an error message:
+\begin{caml_example}{toplevel}[error]
+Printf.printf "Float value: %F" 42;;
+\end{caml_example}
+"fprintf" function is like "printf" except that it takes an output channel as
+the first argument.
+\begin{caml_example}{toplevel}
+let pp_int ppf n = Printf.fprintf ppf "%d" n
+let pp_option printer ppf = function
+  | None -> Printf.fprintf ppf "None"
+  | Some v -> Printf.fprintf ppf "Some(%a)" printer v
+let () = Printf.fprintf stdout
+  "The current setting is %a"
+  (pp_option pp_int) (Some 3);
+print_newline ();;
+\end{caml_example}
+In the "pp_int" we create a printing template that converts an integer
+argument to signed decimal.
+"pp_option" is a function that processes the argument passed to it,
+depending on its value (displays the number if it is not None using a printer
+passed as an argument).
+This approach works for more complex types, for example options. In such case
+we need to provide a printer for any additional type that they wrap.
+In this way we coustruct the composable printer that can be used with the "%a"
+specifier, calling it in the last step of an example.
+
+Here is how to rewrite the pretty-printer using "fprintf":
+\begin{caml_example}{toplevel}
+let print_expr ppf expr =
```

Comment:
`print_expr` could also be renamed to `pp_expr` for consistency, since it could itself be used with "%a" as a composable printer!